sdk/eventhub/event-hubs/src/receiver.ts

```diff
@@ -160,25 +160,58 @@ export class EventHubConsumer {
   receive(onMessage: OnMessage, onError: OnError, abortSignal?: AbortSignalLike): ReceiveHandler {
     this._throwIfReceiverOrConnectionClosed();
     this._throwIfAlreadyReceiving();
+
+    // _throwIfReceiverOrConnectionClosed _should_ cover this case already, but throw an error if something unexpected happened.
+    if (!this._baseConsumer) {
+      const errorMessage =
+        `The EventHubConsumer for "${this._context.config.entityPath}" can no longer be used. ` +
+        `Please create a new EventHubConsumer using the "createConsumer" function on the EventHubClient.`;
+      throw new Error(errorMessage);
+    }
+
     if (typeof onMessage !== "function") {
       throw new TypeError("The parameter 'onMessage' must be of type 'function'.");
     }
     if (typeof onError !== "function") {
       throw new TypeError("The parameter 'onError' must be of type 'function'.");
     }
-    const checkpoint = this.getCheckpoint();
-    if (checkpoint) {
-      this._eventPosition = EventPosition.fromSequenceNumber(checkpoint);
+
+    // return immediately if the abortSignal is already aborted.
+    if (abortSignal) {
+      if (abortSignal.aborted) {
+        onError(new AbortError("The receive operation has been cancelled by the user."));
+        return new ReceiveHandler(this._baseConsumer);
+      }
+
+      abortSignal.addEventListener("abort", () => {
+        if (this._baseConsumer) {
+          this._baseConsumer.abort();
+        }
+      });
+    }
+
+    this._baseConsumer.prefetchCount = Constants.defaultPrefetchCount;
+    if (!this._baseConsumer.isOpen()) {
+      this._baseConsumer
+        .initialize()
+        .then((): any => {
+          if (!this._baseConsumer) {
+            return;
+          }
+          this._baseConsumer.registerHandlers(onMessage, onError, abortSignal);
+          if (abortSignal && abortSignal.aborted) {
+            return this._baseConsumer.abort();
+          }
```

Comment:
Ah, I remember why I did this now. `registerHandlers` will register the error handler right away. I was relying on the baseConsumer to throw the `AbortError`. By flipping these I have to create the `AbortError` inline here as well.

I'll make this change since it seems more clear to future devs (myself included) what the intent is.